esp/src/eclwatch/HexViewWidget.js

```diff
@@ -87,74 +84,45 @@ define([
-                                    arrayUtil.forEach(results, function (result, idx) {
-                                        var store = result.getStore();
-                                        var result = store.query({
-                                        }, {
-                                                start: 0,
-                                                count: context.bufferLength
-                                            }).then(function (response) {
-                                                context.watchHandle.unwatch();
-                                                context.cachedResponse = response;
-                                                context.displayHex();
-                                                context.wu.doDelete();
-                                            });
-                                    });
-                                });
+                this.wu = hpccComms.Workunit.submit({ baseUrl: "" }, "hthor", context.getQuery()).then(function (wu) {
+                    wu.on("changed", function () {
+                        if (!wu.isComplete()) {
+                            context.hexView.setText("..." + wu.State + "...");
+                        }
+                    });
+                    return wu.watchUntilComplete();
+                }).then(function (wu) {
+                    return wu.fetchECLExceptions().then(function () { return wu; });
+                }).then(function (wu) {
+                    var exceptions = wu.Exceptions && wu.Exceptions.ECLException ? wu.Exceptions.ECLException : [];
+                    if (exceptions.length) {
+                        var msg = "";
+                        arrayUtil.forEach(exceptions, function (exception) {
+                            if (exception.Severity === "Error") {
+                                if (msg) {
+                                    msg += "\n";
+                                }
+                                msg += exception.Message;
                             }
-                            break;
-                        case "State":
-                            context.hexView.setText("..." + (context.wu.isComplete() ? context.i18n.fetchingresults : newValue) + "...");
-                            break;
+                        });
+                        if (msg) {
+                            dojo.publish("hpcc/brToaster", {
+                                Severity: "Error",
+                                Source: "HexViewWidget.remoteRead",
+                                Exceptions: [{ Source: context.wu.Wuid, Message: msg }]
+                            });
+                        }
                     }
+                    return wu.fetchResults().then(function (results) {
+                        return results.length ? results[0].fetchRows() : [];
+                    }).then(function (rows) {
+                        context.cachedResponse = "";
```

Comment:
Not sure why the reassignment of type is happening here unless rows is a string as well?

Reply:
Yup - looks wrong.

Reply:
Removed spurious line.